src/index.ts

```diff
@@ -44,3 +45,62 @@ export function getServiceData(apiName: string, beaconName: string, chain: strin
     },
   };
 }
+
+interface PrivateKeySender {
+  privateKey: string;
+}
+
+interface MnemonicSender {
+  mnemonic: string;
+  derivationPath?: string;
+}
+
+type SenderAccount = PrivateKeySender | MnemonicSender;
+
+export function isPrivateKeySender(senderAccount: SenderAccount): senderAccount is PrivateKeySender {
+  return Object.prototype.hasOwnProperty.call(senderAccount, 'privateKey');
+}
+
+interface WhitelistBeaconReaderResult {
+  beaconId: string;
+  beaconReaderAddress: string;
+  expirationTimestamp: number;
+  indefiniteWhitelistStatus: boolean;
+}
+
+export async function whitelistBeaconReader(
+  beaconId: string,
+  beaconReaderAddress: string,
+  chain: string,
+  providerUrl: string,
+  senderAccount: SenderAccount
+): Promise<WhitelistBeaconReaderResult> {
+  const goBeaconWhitelister = goSync(() =>
```

Comment:
I see you really like `goSync()` 😄

Reply:
Yeah, I've said it to @andreogle in my first days that I like that abstraction.